vspace/src/main/java/edu/asu/diging/vspace/web/staff/SpaceOverviewController.java

```diff
@@ -0,0 +1,246 @@
+                                    : null;
+            if (spaceId != null) {
+                if (!spaceToAllOtherLinksMap.containsKey(spaceId)) {
+                    spaceToAllOtherLinksMap.put(spaceId, new ArrayList<>());
+                }
+                spaceToAllOtherLinksMap.get(spaceId).add(link);
+            }
+        });
+    }
+
+    /**
+     * creating json for space and module node in the spaceoverview graph
+     * 
+     * @param <T>
+     * @param request
+     * @param nodeList
+     * @param mapper
+     * @param nodeArray
+     * @param allNodeList
+     * @param isModule
+     */
+    private <T> void nodeFormation(String contextPath, Iterable<T> nodeList, ObjectMapper mapper, ArrayNode nodeArray,
+            List<String> allNodeList, boolean isModule) {
+
+        for (T nodeval : nodeList) {
+            VSpaceElement nodeObj = isModule == true ? (Module) nodeval : (Space) nodeval;
+            if (nodeObj != null) {
+                ObjectNode node = mapper.createObjectNode();
+                node.put("name", nodeObj.getName());
+                StringBuilder linkPathBuilder = new StringBuilder();
+                linkPathBuilder.append(contextPath);
+                String path = isModule == true ? ModuleController.STAFF_MODULE_PATH : SpaceController.STAFF_SPACE_PATH;
+                linkPathBuilder.append(path);
+                linkPathBuilder.append(nodeObj.getId());
+                node.put("link", linkPathBuilder.toString());
+                String imagePath = "";
+                if (!isModule) {
+                    StringBuilder imagePathBuilder = new StringBuilder();
+                    imagePathBuilder.append(contextPath);
+                    imagePathBuilder.append(ImageApiController.API_IMAGE_PATH);
+                    imagePathBuilder.append(((Space) nodeObj).getImage().getId());
+                    imagePath = imagePathBuilder.toString();
+                }
+                node.put("img", imagePath);
+                node.put("isModule", isModule);
+                if (!isModule) {
+                    if (((Space) nodeObj).getSpaceStatus() != null
+                            && ((Space) nodeObj).getSpaceStatus().equals(SpaceStatus.UNPUBLISHED)) {
+                        node.put("isUnpublished", true);
+                    } else {
+                        node.put("isUnpublished", false);
+                    }
+                    if (((Space) nodeObj).isHideIncomingLinks()) {
+                        node.put("isHideIncomingLinks", true);
+                    } else {
+                        node.put("isHideIncomingLinks", false);
+                    }
+                }
+                if (node != null) {
+                    nodeArray.add(node);
+                }
+                if (nodeObj.getId() != null) {
+                    allNodeList.add(nodeObj.getId());
+                }
+            }
+        }
+    }
```

Comment:
to make all of this easier to read and maintain, I would create create two methods instead of this one: one that handles modules, the other that handles spaces; then I would take all the common parts and put those into shared methods, so that the modules and spaces methods just call other common methods but do what is specific to their classes, eg.:
```
// spaces method
for (Space space : spaces) {
   buildCommonNodeProperties(...) // also passing path to be used in link
   // set image path
   // set unpublished property
  // etc
}
```
similar for modules.

Does that make sense?